syntax = "proto3";

service Master {
    rpc RegisterWorker(RegisterWorkerRequest) returns (RegisterWorkerResponse) {}
    rpc SendHeartbeat(SendHeartbeatRequest) returns (SendHeartbeatResponse) {}
    rpc TaskCompletion(TaskCompletionRequest) returns (TaskCompletionResponse) {}
}

message RegisterWorkerRequest {
    int32 worker_id = 1;
}

message RegisterWorkerResponse {
    bool success = 1;
}

message SendHeartbeatRequest {
    int32 worker_id = 1;
}

message SendHeartbeatResponse {
    bool success = 1;
}

message TaskCompletionRequest {
    int32 worker_id = 1;
    int32 task_id = 2;
}

message TaskCompletionResponse {
    bool success = 1;
}